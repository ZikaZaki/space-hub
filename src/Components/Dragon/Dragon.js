import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { toggleDragon } from '../../redux/dragons/dragons';
// import styles
import styles from './Dragon.module.css';

const Dragon = ({
  id,
  name,
  description,
  image,
  reserved,
}) => {
  const dispatch = useDispatch();

  return (
    <div className={styles['dragon-container']}>
      <img className={styles['dragon-img']} src={image} alt="dragon" />
      <div className={styles['dragon-info']}>
        <div className={styles['dragon-name']}>
          <h2>{name}</h2>
          {reserved && <span className={styles.badge}>Reserved</span>}
        </div>
        <p>{description}</p>
        {reserved ? <button type="button" className={styles['cancel-btn']} onClick={() => dispatch(toggleDragon(id))}>Cancel Reservation</button>
          : <button type="button" className={styles['reserve-btn']} onClick={() => dispatch(toggleDragon(id))}>Reserve Dragon</button>}
      </div>
    </div>
  );
};

Dragon.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Dragon;
