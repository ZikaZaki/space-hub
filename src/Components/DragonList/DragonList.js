import React from 'react';
import Dragon from '../Dragon/Dragon';
// import styles
import styles from './DragonList.module.css';

const DragonList = () => (
  <div className={styles['dragon-list-container']}>
    <ul className={styles['dragon-list']}>
      <li className={styles['dragon-item']}>
        <Dragon />
      </li>
      <li className={styles['dragon-item']}>
        <Dragon />
      </li>
      <li className={styles['dragon-item']}>
        <Dragon />
      </li>
    </ul>
  </div>
);

export default DragonList;
