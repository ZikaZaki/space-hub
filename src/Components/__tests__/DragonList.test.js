import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Dragon from '../Dragon/Dragon';
import DragonList from '../DragonList/DragonList';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <DragonList>
        <Dragon
            id="dragon1"
            name="Dragon 1"
            description="Dragon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009"
            image="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
            reserved={false}
        />
      </DragonList>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render a div container', () => {
  render(
    <Provider store={store}>
      <DragonList>
        <Dragon
            id="dragon1"
            name="Dragon 1"
            description="Dragon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009"
            image="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
            reserved={false}
        />
      </DragonList>
    </Provider>,
  );

  const dragon = screen.getByTestId('dragon-list-container');

  expect(dragon).toBeInTheDocument();
});
