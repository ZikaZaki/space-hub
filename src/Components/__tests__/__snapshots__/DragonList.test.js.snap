// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component renders correctly 1`] = `
<div
  className="dragon-list-container"
  data-testid="dragon-list-container"
>
  <ul
    className="dragon-list"
  />
</div>
`;
